var/spack/repos/builtin/packages/dyninst/package.py

```diff
@@ -81,7 +81,7 @@ class Dyninst(CMakePackage):
         sha256="0064d8d51bd01bd0035e1ebc49276f627ce6366d4524c92cf47d3c09b0031f96",
     )
 
-    requires("%gcc", when="@:13.0.0", msg="dyninst builds only with GCC")
+    requires("%gcc", when="@:12", msg="dyninst builds only with GCC")
```

Comment:
The last release before 13.0 was 12.3.

```suggestion
    requires("%gcc", when="@:12.3", msg="dyninst builds only with GCC")
```

Reply:
Actually `@:12` includes `@12.3.0` :)

Reply:
Oh, I didn't realize they were fuzzy-matched like that. So `@:12.3` would also match `12.3.1`?

Reply:
Yes, that's right! Admittedly I didn't know either, but I looked it up to make this PR :). I thought upper bound on version ranges were exclusive, so I was surprised when the requirement for %gcc was being applied to @13.0.0